extensions/resource-deployment/src/test/services/notebookService.test.ts

```diff
@@ -0,0 +1,294 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import 'mocha';
+import * as path from 'path';
+import * as should from 'should';
+import * as sinon from 'sinon';
+import * as TypeMoq from 'typemoq';
+import * as vscode from 'vscode';
+import { NotebookPathInfo } from '../../interfaces';
+import { Notebook, NotebookService } from '../../services/notebookService';
+import { IPlatformService 	} from '../../services/platformService';
+import * as loc from '../../localizedConstants';
+import assert = require('assert');
+
+class Deferred<T> {
+	promise: Promise<T>;
+	resolve!: (value?: T | PromiseLike<T>) => void;
+	reject!: (reason?: any) => void;
+	constructor() {
+		this.promise = new Promise<T>((resolve, reject) => {
+			this.resolve = resolve;
+			this.reject = reject;
+		});
+	}
+}
+describe('NotebookService', function (): void {
+	const notebookInput = 'test-notebook.ipynb';
+	const sourceNotebookContent = '{ "cells": [] }';
+	const notebookFileName = 'mynotebook.ipynb';
+	const expectedTargetFileName = 'mynotebook';
+	const extensionPath = path.resolve(__dirname, '..', '..', '..');
+	const sourceNotebookRelativePath = `./notebooks/${notebookFileName}`;
+	const sourceNotebookAbsolutePath = path.resolve(extensionPath, sourceNotebookRelativePath);
+	const notebookWin32 = 'test-notebook-win32.ipynb';
+	const notebookDarwin = 'test-notebook-darwin.ipynb';
+	const notebookLinux = 'test-notebook-linux.ipynb';
+	const storagePath = __dirname;
+	let mockPlatformService: TypeMoq.IMock<IPlatformService>, notebookService: NotebookService;
+
+	beforeEach('NotebookService Setup', () => {
+		mockPlatformService = TypeMoq.Mock.ofType<IPlatformService>();//PlatformService, TypeMoq.MockBehavior.Loose
+		notebookService = new NotebookService(mockPlatformService.object, extensionPath);
+	});
+
+	afterEach('NotebookService cleanup', () => {
+		sinon.restore();
+		mockPlatformService.reset();
+	});
+
+	it('getNotebook with string parameter', async () => {
+		mockPlatformService.setup(x => x.fileExists(notebookInput)).returns(async () => true);
+		mockPlatformService.setup(x => x.readTextFile(notebookInput)).returns(async () => sourceNotebookContent);
+		let returnValue = await notebookService.getNotebook(notebookInput);
+		assert.deepEqual(returnValue, JSON.parse(sourceNotebookContent), 'returned notebook does not match expected value');
```

Comment:
Please use should assertions - the output error messages are typically a lot better